decentraland/redux/reducers.js

```diff
@@ -77,6 +78,10 @@ function myAssets(state = { list: [] }, action) {
       const list = removeFromList(myAssets, myAsset);
       return { ...state, list };
     }
+    case SET_MY_ASSETS_LIST: {
+      const list = myAssets === null ? [] : myAssets;
```

Comment:
I wonder whether we want to ensure that the `myAssets` state variable as used via the UI isn't `null` in this case too - not just when hitting storage via this reducer

Reply:
Not sure if I've got it.
On the way that it is now `myAssets.list` never will be null. It starts as `[]`, both `add` and `remove` actions will keep it as an array and the `set` is checking it.
Oh, worth check if `undefined` or non-array values are being passed?

Reply:
I guess my question is why `myAssets` would ever be null when passed into this reducer

Reply:
Sorry, you meant "ever" or "never"?

Reply:
I meant "ever".

In other words, I think that we should make sure that the data we pass to this reducer is either an empty array or a non-empty array, but never null.

Reply:
Oh, yes, I agree. So do you think will be better to remove that validation from here?

Reply:
Well, it's not even validation that we're doing here as it's written. There may even be a missing `.list` based on the way you described it above?

How I read it now is, sometimes we'll pass in a null value for myAssets. If that happens, pretend we passed in [] instead. Otherwise, use myAssets (not myAssets.list)

Reply:
I've putted back the checks on `App.js`, we can change `retrieveMyAssets` functions to return `[]` instead of `null` too, anyway I often put that kind of validation closest to persistance to protected data from a buggy input code.

Reply:
> Well, it's not validation that we're doing here as it's written. There may be a missing `.list` based on the way you described it above?
> 
> How I read it now is, sometimes we'll pass in a null value for myAssets. If that happens, pretend we passed in [] instead. Otherwise, use myAssets (not myAssets.list)

Yes, maybe we should rename vars. The action passes an array called `myAssets`, the same name as the `reducer`, but the reducer has that structure `{ list: [] }`.
So, if action sends a null, the code will stores an `[]` instead.

Reply:
Ah okay yeah, I think what I was wanting what changing `retrieveMyAssets` to return `[]` instead of `null`.

Now I think I get what you were saying - I wasn't always making sense in this thread. What we're doing here is enforcing valid data in persistence in a sense, and I like the idea of enforcing that at the reducer level. So I changed my mind, this line should stay as-is.

Reply:
Or wait, retrieving the assets happens from the data store or the blockchain?

Retrieving from the datastore will automatically never be null if we enforce this on the way in.

The function we may want to change is the one that gets data from the blockchain.

Reply:
retrieveMyAssets changed

Reply:
> Retrieving from the datastore will automatically never be null if we enforce this on the way in.

This isn't totally true actually - it would be null if it has never been set.

Reply:
> Or wait, the retrieving the assets done by `retrieveMyAssets` is getting it from the data store or the blockchain?
> 
> Retrieving from the datastore will automatically never be null if we enforce this on the way in.
> 
> The function we may want to change is the one that gets data from the blockchain.

retrieveMyAssets is getting data from store, if has no store yet it was returning `null`, now `[]`.
I'll make that caution on the next PR (when I'll get data from `DecentralandUtils`)

Reply:
> The function we may want to change is the one that gets data from the blockchain.

This one should return `[]` when there are no assets. But that's coming up in the next PR

Reply:
All your responses make sense and sound good. We should rename `retrieveMyAssets` to indicate it's retrieving them from storage too.

Reply:
I'm going to merge this PR as-is